aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -1233,6 +1259,9 @@ void TensorIterator::build(TensorIteratorConfig& config) {
 
   // fill in operands_ based on configuration
   populate_operands(config);
+  if (is_reduction_ && num_outputs_ > 1){
+    create_reduction_temporaries();
```

Comment:
how do you know the shapes are correct?

Reply:
all the reductions go through `make_reduction` that allocates correctly sized outputs. It's a good point though that if someone were to call TensorIterator::reduce_op directly, or manually build TensorIteratorConfig with the wrong-sized outputs that would lead to an error, but it's also true today.

Reply:
does putting this after `compute_types` or something not fix this specific issue though?

Reply:
Which issue, `outputs` potentially being wrong size? I don't think TensorIterator ever checks that outputs are correct size for reduction.

Reply:
oh that's good.